src/Core/Components/List/ListComponentBase.razor.cs

```diff
@@ -538,6 +543,11 @@ protected virtual async Task OnSelectedItemChangedHandlerAsync(TOption? item)
                 RemoveSelectedItem(item);
                 await RaiseChangedEventsAsync();
             }
+            else if (OptionComparer is not null && _selectedOptions.FirstOrDefault(x => OptionComparer.Equals(x, item)) is TOption addedItem)
```

Comment:
Can you add a test `OptionComparer is null` from the previous `if` and use `Find` (and not `FirstOrDefault` (for better performances)

```csharp
if (OptionComparer is null && _selectedOptions.Contains(item))
{
    RemoveSelectedItem(item);
    await RaiseChangedEventsAsync();
}
else if (OptionComparer is not null && _selectedOptions.Find(x => OptionComparer.Equals(x, item)) is TOption addedItem)
{
    RemoveSelectedItem(addedItem);
    await RaiseChangedEventsAsync();
}
```

Reply:
done